### PR TITLE
More than tickets

### PR DESCRIPTION
**Pull request by sinkingsugar**

### CODE REVIEW

**Comment by sinkingsugar** [comment]
Going to merge this ASAP as it's needed by one of our customers and our testnet bot.
Please have a look!

File: docs/Pallets Dependency Graph.excalidraw.md [hunk 1]
```diff
@@ -0,0 +1,1100 @@
+---
```
Comment (aang114) [inline docs/Pallets Dependency Graph.excalidraw.md:1]: what is this file even supposed to be ? 😓

File: pallets/fragments/src/lib.rs [hunk 2]
```diff
@@ -274,8 +274,12 @@ pub mod pallet {
 			// create vault account
 			// we need an existential amount deposit to be able to create the vault account
 			let vault = Self::get_vault_id(hash);
-			let min_balance = T::Currency::minimum_balance();
```
Comment (aang114) [inline pallets/fragments/src/lib.rs:277]: What's the point of have the assosciated type `Currency` for the `Config` trait of `pallet_assets` anymore?

Comment (aang114) [reply pallets/fragments/src/lib.rs:277]: anyways, this is not even a bug - just something redundant

File: pallets/accounts/src/lib.rs [hunk 3]
```diff
@@ -496,6 +517,52 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Update 'data'
+		///
+		/// TODO
+		#[pallet::weight(25_000)] // TODO #1 - weight
+		pub fn sponsor_account(origin: OriginFor<T>, external_id: ExternalID) -> DispatchResult {
```
Comment (alebaffa) [inline pallets/accounts/src/lib.rs:525]: Looks good. We will document this later.

**Review by alebaffa — APPROVED** [review]
LGTM 👍🏻

File: pallets/accounts/src/lib.rs [hunk 4]
```diff
@@ -496,6 +517,52 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Update 'data'
+		///
+		/// TODO
+		#[pallet::weight(25_000)] // TODO #1 - weight
```
Comment (aang114) [inline pallets/accounts/src/lib.rs:524]: why not call this extrinsic `link_account` instead of `sponsor_account`?

Comment (aang114) [reply pallets/accounts/src/lib.rs:524]: this function looks correct.

But just to confirm, when the Linked External User Account (for e.g the Discord Account) wants to use the Linked External User Account to call the extrinsic `<EXTRINSIC HE WANTS TO CALL>` - 

he will have to call the extrinsic: 

`pallet_proxy::Pallet::proxy(<CLAMOR ACCOUNT ID THAT IS LINKED TO THE ANON PROXY>, <THE ANON PROXY>, T::ProxyType, <EXTRINSIC HE WANTS TO CALL>)`

is this correct @sinkingsugar?

File: pallets/accounts/src/lib.rs (hunk 3 above)
Comment (sinkingsugar) [reply pallets/accounts/src/lib.rs:525]: Yeah sorry, this is what the discord bot uses 👯

**Review by aang114 — APPROVED** [review]
LGTM

**Comment by alebaffa** [comment]
@sinkingsugar The branch is failing because of this:
```rust
| impl Config for Test {
    |      ^^^^^^ the trait `pallet_proxy::Config` is not implemented for `Test`
```

I think this can fix it. To be added in both `pallets/accounts/src/mock.rs` and `pallets/fragments/src/mock.rs` (though, I am not sure about all the types assignments in the `impl `: 
```rust
frame_support::construct_runtime!(
    pub enum Test where
    ....
    {
        ....
        Proxy: pallet_proxy::{Pallet, Call, Event<T>, Storage},
    }

impl pallet_proxy::Config for Test {
    type Event = Event;
    type Call = Call;
    type Currency = ();
    type ProxyType = ();
    type ProxyDepositBase = ();
    type ProxyDepositFactor = ();
    type MaxProxies = ConstU32<32>;
    type WeightInfo = ();
    type MaxPending = ConstU32<32>;
    type CallHasher = BlakeTwo256;
    type AnnouncementDepositBase = ();
    type AnnouncementDepositFactor = ();
}
```

**Review by alebaffa — CHANGES_REQUESTED** [review]
@sinkingsugar The branch is failing because of this:

```rust
| impl Config for Test {
    |      ^^^^^^ the trait `pallet_proxy::Config` is not implemented for `Test`
```
I think this can fix it. To be added in both `pallets/accounts/src/mock.r`s and `pallets/fragments/src/mock.rs` (In Substrate [node/runtime/src/lib.rs](https://github.com/paritytech/substrate/blob/master/bin/node/runtime/src/lib.rs) there is an example of usage of pallet_proxy. Can be useful to see how to implement all the types needed) :

```rust
frame_support::construct_runtime!(
    pub enum Test where
    ....
    {
        ....
        Proxy: pallet_proxy::{Pallet, Call, Event<T>, Storage},
    }

impl pallet_proxy::Config for Test {
    type Event = Event;
    type Call = Call;
    type Currency = ();
    type ProxyType = ();
    type ProxyDepositBase = ();
    type ProxyDepositFactor = ();
    type MaxProxies = ConstU32<32>;
    type WeightInfo = ();
    type MaxPending = ConstU32<32>;
    type CallHasher = BlakeTwo256;
    type AnnouncementDepositBase = ();
    type AnnouncementDepositFactor = ();
}
```

**Review by alebaffa — APPROVED** [review]

**Pull request merged**